esp_utils.py

```
@@ -51,3 +51,25 @@ def check_esp_idf_installed() -> bool:
+    try:
+        # Try to use idf.py to list ports (if available)
+        process = await asyncio.create_subprocess_shell(
+            "python -m serial.tools.list_ports",
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE
+        )
+        stdout, stderr = await process.communicate()
+        return process.returncode, stdout.decode(), stderr.decode()
+    except Exception as e:
+        # Fallback: try common port patterns
+        common_ports = ["/dev/ttyUSB0", "/dev/ttyUSB1", "/dev/ttyACM0", "/dev/ttyACM1", 
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)